src/reducers/consoles.js

```diff
@@ -1,50 +1,54 @@
-import Immutable from 'immutable'
+// @flow
+
+import * as Immutable from 'immutable'
 import {
   SET_CONSOLE_TICKETS,
-  SET_CONSOLE_NOVNC_STATUS,
-  SET_NEW_CONSOLE_MODAL,
-  CLOSE_CONSOLE_MODAL,
-  SET_IN_USE_CONSOLE_MODAL_STATE,
-  SET_LOGON_CONSOLE_MODAL_STATE,
-
-  CONSOLE_OPENED,
-  CONSOLE_IN_USE,
-  CONSOLE_LOGON,
+  SET_CONSOLE_STATUS,
+  ADD_CONSOLE_ERROR,
+  DISMISS_CONSOLE_ERROR,
 } from '_/constants'
 
 import { actionReducer } from './utils'
-import { SET_ACTIVE_CONSOLE } from '../constants'
+import type { ConsoleErrorType, VmConsoleType } from '_/ovirtapi/types'
 
-const initialState = Immutable.fromJS({ vms: {}, modals: {} })
+const initialState: {
+  errors: Array<ConsoleErrorType>,
```

Comment:
Is this really an array of errors?  I don't remember seeing it being considered as an array.  The `ConsoleNotificationsDialog` does `consoles.getIn(['errors', 0]`....

Reply:
yes. There can be (theoretically) multiple errors (from different consoles and VMs)

Reply:
So then `ConsoleNotificationsDialog` needs to be looking at the correct error right instead of just `[0]` right?

Reply:
`errors` are used as a queue here: the dialog displays the first error in the queue (if any).

Reply:
Just a bit confusing to keep track of things pushing data into errors and how they get iterated over to display (the dialog dismisses one each time it is rendered).